enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/JWTAuthenticator.java

```diff
@@ -411,6 +414,17 @@ private void checkTokenEnvAgainstDeploymentType(String keyType, APIConfig matche
         if (System.getenv("DEPLOYMENT_TYPE_ENFORCED") != null
                 && System.getenv("DEPLOYMENT_TYPE_ENFORCED").equalsIgnoreCase("false")
                 && keyType.equalsIgnoreCase(APIConstants.JwtTokenConstants.PRODUCTION_KEY_TYPE)) {
+            if (System.getenv("WHITELISTED_ORGANIZATIONS") != null) {
+                for (String whitelistedOrgId : whitelistedOrganizationsArray) {
+                    if (matchedAPI.getOrganizationId().equalsIgnoreCase(whitelistedOrgId)) {
+                        return;
+                    }
+                }
+                throw new APISecurityException(APIConstants.StatusCodes.UNAUTHORIZED.getCode(),
+                        APISecurityConstants.API_AUTH_INVALID_ENVIRONMENT,
+                        APISecurityConstants.API_AUTH_INVALID_ENVIRONMENT_ERROR_MESSAGE);
+            }
+
             log.info("Deprecated: Production access token is used to access sandbox API deployment in " +
                     "organization : " +  matchedAPI.getOrganizationId());
```

Comment:
We still have to print this warning despite the configuration. We need to identify which orgs are accessing like this even with the configuration. We need to get rid of everyone eventually.

Reply:
Since the log is present outside the if condition to check for the env variable, it still prints the log irrespective of the config

Reply:
Problem is when the config is enabled.
There's a return inside the block.

Reply:
Moved the log block